### DISCUSSION TITLE
Как изолировать выполняемые процессы?

### DISCUSSION THREAD

**Discussion by gtf0me** [totumonline/totum-ru-issues-and-discussions#314] [Как это сделать?] [answered]
Привет! Есть ли техническая возможность оптимизировать процесс:

1. Принять запрос на Remotes
2. Изменить значение в таблице в какой-либо ячейке на $#get[param]
3. Код действия на изменение ячейки в таблице: записать строку в другую таблицу
4. Другая таблица при добавлении данных: отправляет вебхук, если ответ с того конца приходит не "success = true", то ждёт минуту и отправляет запрос снова (и так далее до бесконечности)

Тотумом это всё воспринимается, как одно большое действие и когда выполняешь шаг 1, то в итоге будешь ждать вечность, пока отработает шаг 4 до ответа "success = true". 
Вопрос: как можно сделать, чтобы шаг 1, 2 выполнялся, Remotes возвращал ответ клиенту, а шаги 3, 4 запускались уже без участия висящего и ожидающего ответ клиента на Remotes?

**Comment by erpic-ru** [comment, marked as answer]
Привет! Через функцию [exec](https://ru.docs.totum.online/functions#fn-exec) с параметром ssh:true? 

https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/58
https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/161

**Reply by ttm-alexeypolunin** [reply]
Если там в эту таблицу не только ремоут добавляет, а отделять нужно только для ремоута, то хорошо бы использовать процессную переменную, что бы знать, что вызывает ремоут и если она например `true`, то тогда отделять в отдельный процесс.

Процессные переменные — они изолированы между процессами (те если два одинаковых ремоута запустятся, то процессные переменные с одинаковым именем в них не будет пересекаться) и еще они автоматически удаляются при завершении процесса.

Как это будет выглядеть.

В ремоуте мы создаем переменную и запускаем `insert` в таблицу:

```
=: while(action: $pvar; action: $insert)

pvar: procVar(name: "ssh_remote"; value: true)

insert: insert(...)
```

В коде при добавлении в целевой таблице мы проверяем переменную и если она true запускаем ssh, если нет, то просто выполняем:

```
=: exec(code: $code; ssh: $@ssh_remote)

    ```code:totum

    =: "... code here ..."
    ```

```

**Reply by gtf0me** [reply]
Большое спасибо за объяснение!
